Bio/Seq.py

```diff
@@ -852,23 +852,43 @@ def complement(self):
         Note in the above example, ambiguous character D denotes
         G, A or T so its complement is H (for C, T or A).
 
-        Trying to complement a protein sequence raises an exception.
+        Note that if the sequence contains neither T nor U, we
+        assume it is DNA and map any A character to T:
+
+        >>> Seq("CGA").complement()
+        Seq('GCT')
+        >>> Seq("CGAT").complement()
+        Seq('GCTA')
+        >>> Seq("CGAU").complement()
+        Seq('GCUA')
+
+        If you are dealing with RNA which might have no U, use:
+
+        >>> Seq("CGA").complement().transcribe()
+        Seq('GCU')
+
+        If the sequence contains both T and U, an exception is
+        raised:
+
+        >>> Seq("CGAUT").complement()
+        Traceback (most recent call last):
+           ...
+        ValueError: Mixed RNA/DNA found
+
+        Trying to complement a protein sequence gives a meaningless
+        sequence:
 
         >>> from Bio.Alphabet import generic_protein
         >>> my_protein = Seq("MAIVMGR", generic_protein)
         >>> my_protein.complement()
-        Traceback (most recent call last):
-           ...
-        ValueError: Proteins do not have complements!
+        Seq('KTIBKCY')
+
+        Here "M" was interpretted as the IUPAC ambiguity code for
+        "A" or "C", with complement "K" for "T" or "G". Likewise
+        "A" has complement "T". The letter "I" has not defined
+        meaning under the IUPAC convention, and is unchanged.
         """
-        base = Alphabet._get_base_alphabet(self.alphabet)
-        if isinstance(base, Alphabet.ProteinAlphabet):
-            raise ValueError("Proteins do not have complements!")
-        if isinstance(base, Alphabet.DNAAlphabet):
-            ttable = _dna_complement_table
-        elif isinstance(base, Alphabet.RNAAlphabet):
-            ttable = _rna_complement_table
-        elif ("U" in self._data or "u" in self._data) and (
+        if ("U" in self._data or "u" in self._data) and (
             "T" in self._data or "t" in self._data
         ):
```

Comment:
Perhaps `.complement` should always assume the sequence is DNA (and raise and error otherwise). We could add a separate method `.rna_complement` for RNA.

Reply:
There is merit in that... especially as the majority of usage will be DNA not RNA, and an additional argument would be messy. I will ponder that.